store/src/prune_list.rs

```diff
@@ -179,6 +180,18 @@ impl PruneList {
 		}
 	}
 
+	// Calculate the next shift based on provided pos and the previous shift.
+	fn calculate_next_shift(&self, pos: u64) -> u64 {
+		let prev_shift = self.get_shift(pos.saturating_sub(1));
```

Comment:
pos.saturating_sub(1) should be simply pos-1 since pos is 1-based and cannot be 0.

Reply:
I suggest we change all occurrences of pos to either pos0 or pos1 depending on whether they're 0-based or 1-based.

Reply:
In this case yes, we can just change the saturating sub since it's only called in one place at the moment that will fail with assertion beforehand is pos is 0

Reply:
I'm going to do a bit of a 'clean as you go' on changing pos to pos0 or pos1 at the moment, but if we do a complete pass on all instances it should be within a separate PR.

Reply:
Let me know when you're done making pmmr changes so I can do my own cleanup...